nexus/db-queries/src/db/datastore/inventory.rs

```diff
@@ -3226,6 +3263,31 @@ impl DataStore {
             memberships
         };
 
+        // Load the cockroach status records for all nodes.
+        let cockroach_status: BTreeMap<CockroachNodeId, CockroachStatus> = {
+            use nexus_db_schema::schema::inv_cockroachdb_status::dsl;
+
+            let status_records: Vec<InvCockroachStatus> =
+                dsl::inv_cockroachdb_status
+                    .filter(dsl::inv_collection_id.eq(db_id))
+                    .select(InvCockroachStatus::as_select())
+                    .load_async(&*conn)
+                    .await
+                    .map_err(|e| {
+                        public_error_from_diesel(e, ErrorHandler::Server)
+                    })?;
+
+            let mut result = BTreeMap::new();
+            for record in status_records {
+                let node_id = CockroachNodeId::new(record.node_id);
+                let status: nexus_types::inventory::CockroachStatus = record
+                    .try_into()
+                    .map_err(|e| Error::internal_error(&format!("{e:#}")))?;
+                result.insert(node_id, status);
+            }
+            result
```

Comment:
nit, take it or leave it: isn't this equivalent to
```suggestion
            status_records.into_iter()
                .map(|record| {
                    let node_id = CockroachNodeId::new(record.node_id);
                    let status: nexus_types::inventory::CockroachStatus = record
                        .try_into()
                        .map_err(|e| Error::internal_error(&format!("{e:#}")))?;
                    Ok((node_id, status))
                })
                .collect::<Result<BTreeMap<_, _>, _>>()
```

which...honestly might not be any clearer...